pandas/tests/frame/test_arithmetic.py

```diff
@@ -1061,6 +1061,30 @@ def test_frame_single_columns_object_sum_axis_1():
     tm.assert_series_equal(result, expected)
 
 
+def test_frame_multi_index_operations():
+    # GH 43321
+    df = DataFrame(
+        {2010: [1, 2, 3], 2020: [3, 4, 5]},
+        index=MultiIndex.from_product(
+            [["a"], ["b"], [0, 1, 2]], names=["scen", "mod", "id"]
+        ),
+    )
+
+    s = Series(
+        [0.4],
+        index=MultiIndex.from_product([["b"], ["a"]], names=["mod", "scen"]),
+    )
+
+    expected = DataFrame(
+        {2010: [1.4, 2.4, 3.4], 2020: [3.4, 4.4, 5.4]},
+        index=MultiIndex.from_product(
+            [["a"], ["b"], [0, 1, 2]], names=["scen", "mod", "id"]
+        ),
+    )
+
+    tm.assert_frame_equal(df.add(s, axis=0), expected)
+
```

Comment:
You mean doesn't align because there are no index levels in `s` that align to `df`.

For example
```python
import pandas as pd
df = pd.DataFrame(
    {2010: [1, 2, 3], 2020: [3, 4, 5]},
    index=pd.MultiIndex.from_product(
        [["a"], ["b"], [0, 1, 2]], names=["scen", "mod", "id"]
    ),
)

s = pd.Series(
    [0.4],
    index=pd.MultiIndex.from_product(
        [["c"], ["a"]], names=["mod", "scen"]
    ),
)

print(df.add(s, axis=0))
```
gives
```
              2010  2020
scen mod id             
a    b   0     NaN   NaN
         1     NaN   NaN
         2     NaN   NaN
     c   NaN   NaN   NaN
```
Is that what you mean?

BTW: I just discovered that `s.add(df)` does not work. I would like to do that in a separate Issue/PR as it is the `axis=1` case of `.align()`. Moreover I am not sure if I can do that during the next days and I would like to finish this one.

Reply:
I added a couple of tests. Please check if it is what you mean.

BTW I discovered that in all the test cases the inverse way `s.add(df, axis=0)` does not work in all the cases. I would like to treat this in a separate PR as it seems to run into the `axis=1` case of `.align()`. I want to keep the PRs small and finished quickly.

Reply:
totally fine. if you can do soonish then pls open a PR, otherwise can open an issue to track.